internal/cache/db.go

```diff
@@ -293,66 +281,64 @@ func openAndInitDB(path, dirtyFlagPath string) (*bbolt.DB, error) {
 
 // cleanExpiredUsers removes from the cache any user that exceeded the maximum amount of days without
 // online authentication.
-func (c *Cache) cleanExpiredUsers(expirationDate time.Time) (err error) {
+func (c *Cache) cleanExpiredUsers(expirationDate time.Time, procDir string) (err error) {
 	defer decorate.OnError(&err, "could not clean up database")
 
-	txError := c.db.Update(func(tx *bbolt.Tx) (txError error) {
+	return c.db.Update(func(tx *bbolt.Tx) (err error) {
 		// Cleans up the database
-		activeUsers, txError := getActiveUsers()
-		if txError != nil {
-			err = txError
-			return nil
+		activeUsers, err := getActiveUsers(procDir)
+		if err != nil {
+			return err
 		}
 
-		buckets, txError := getAllBuckets(tx)
-		if txError != nil {
-			err = txError
-			return nil
+		buckets, err := getAllBuckets(tx)
+		if err != nil {
+			return err
 		}
 
 		var expiredUsers []userDB
-		cursor := buckets[userByIDBucketName].Cursor()
-		for k, v := cursor.First(); k != nil; k, v = cursor.Next() {
+		// The ForEach call does not return an error, so we can ignore it.
+		_ = buckets[userByIDBucketName].ForEach(func(k, v []byte) error {
```

Comment:
I would still capture an error for the day we change the implementation internal, it doesn’t bring us anything to ignore it, no?

Reply:
We'll lose some coverage %, but it's fine I think.